app/src/org/commcare/activities/connect/PersonalIdCredentialActivity.kt

```diff
@@ -1,42 +1,123 @@
-
+import org.commcare.utils.MultipleAppsUtil
 
 class PersonalIdCredentialActivity : AppCompatActivity() {
     private val binding: ActivityPersonalIdCredentialBinding by lazy {
         ActivityPersonalIdCredentialBinding.inflate(layoutInflater)
     }
+    private lateinit var credentialsViewPagerAdapter: CredentialsViewPagerAdapter
+    private lateinit var personalIdCredentialViewModel: PersonalIdCredentialViewModel
+    private var personalIdSessionData: PersonalIdSessionData? = null
+    private var userName: String? = null
+    private var profilePic: String? = null
+    private var installedAppRecords: List<PersonalIdCredential> = emptyList()
+    private val titles = listOf(R.string.earned, R.string.pending)
+    private val icons = listOf(R.drawable.ic_credential_earned, R.drawable.ic_credential_pending)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
-        supportActionBar?.title = getString(R.string.my_credentials)
-        retrieveCredentials()
+        credentialsViewPagerAdapter = CredentialsViewPagerAdapter(this)
+        binding.vpCredentials.adapter = credentialsViewPagerAdapter
+        TabLayoutMediator(binding.tabCredentials, binding.vpCredentials) { tab, position ->
+            tab.text = getString(titles[position])
+            tab.setIcon(icons[position])
+        }.attach()
+        personalIdCredentialViewModel = ViewModelProvider(
+            this, ViewModelProvider.AndroidViewModelFactory.getInstance(application)
+        )[PersonalIdCredentialViewModel::class.java]
+        personalIdSessionData =
+            ViewModelProvider(this)[PersonalIdSessionDataViewModel::class.java].personalIdSessionData
+        userName = personalIdSessionData!!.userName
+        profilePic = personalIdSessionData?.photoBase64
+        personalIdCredentialViewModel.setUserInfo(userName!!, profilePic!!)
+        getInstalledAppsList()
+        callCredentialApi()
+        setUpUi()
+        observeCredentialApiCall()
     }
 
-    fun retrieveCredentials() {
+    private fun getInstalledAppsList() {
+        val usableApps = MultipleAppsUtil.getUsableAppRecords()
+
+        installedAppRecords = usableApps.map { record ->
+            PersonalIdCredential().apply {
+                appId = record.applicationId
+                title = record.displayName ?: ""
+            }
+        }
+    }
 
-        val user = ConnectUserDatabaseUtil.getUser(this)
+    private fun setUpUi() {
+        supportActionBar!!.apply {
+            title = getString(R.string.my_worker_history)
+            setDisplayShowHomeEnabled(true)
+            setDisplayHomeAsUpEnabled(true)
+        }
+    }
 
-        object : PersonalIdApiHandler<PersonalIdValidAndCorruptCredential>() {
-            override fun onSuccess(personalIdValidAndCorruptCredential: PersonalIdValidAndCorruptCredential) {
+    private fun observeCredentialApiCall() {
```

Comment:
confirming that the user is able to see the already stored data if this call fails ?